torch/distributed/fsdp/api.py

```diff
@@ -381,7 +387,21 @@ class LocalOptimStateDictConfig(OptimStateDictConfig):
 
 @dataclass
 class ShardedOptimStateDictConfig(OptimStateDictConfig):
```

Comment:
nothing related to this PR, but i'm in general curious, why we need to have separate `OptimStateDictConfig` and normal `StateDictConfig`? is there anything significantly different than the Module state_dict for the optimizer states dicts that we have to introduce separate config? cc @fegin